### PR TITLE
Feature/mybooks yrg banner

### LINKED ISSUE #7361: Reading Goal Banner bar UX improvement
Clicking "Set your 2023 goal" should take you to /books and highlight the "set 2023 goal!" button.

(Maybe we want it to open the modal directly?)

### PR DESCRIPTION
**Pull request by mekarpeles**
Closes #7361
Closes #7367

Moves the main site banner to a mybooks-page specific banner.

Makes it so clicking on "set me goal" from within the banner triggers modal

on http://ol-dev1.us.archive.org:1337/people/mekBot/books

### CODE REVIEW

File: openlibrary/plugins/openlibrary/js/check-ins/index.js [hunk 1]
```diff
@@ -520,20 +520,26 @@ function addGoalSubmissionListener(submitButton) {
...
+			}
                     } else {
-                        const progressComponent = modal.closest('.reading-goal-progress')
-                        updateProgressComponent(progressComponent, Number(formData.get('goal')))
+                        const progressComponent = document.querySelector('.reading-goal-progress')
+			if (progressComponent) {
+                            updateProgressComponent(progressComponent, Number(formData.get('goal')))
+			}
                     }
+		    location.reload()
```
Comment (jimchamp) [inline openlibrary/plugins/openlibrary/js/check-ins/index.js:542]:
```suggestion
                    location.reload()
```
`npm run lint-fix:js` fixes every the other linting errors, but, for some reason, not this one.

File: static/css/components/page-banner.less [hunk 2]
```diff
@@ -7,13 +7,19 @@
 
 .page-banner {
   display: block;
-  color: @white;
+  color: @black;
```
Comment (jimchamp) [inline static/css/components/page-banner.less:10]:
```suggestion
  color: @white;
```
Black on a dark grey background is very hard to read.

Comment (mekarpeles) [reply static/css/components/page-banner.less:10]: Good catch, this was only supposed to be on `-mybooks`

**Review by jimchamp — CHANGES_REQUESTED** [review]
Apologies for not properly posting the books.html comments last Friday.  The commit linked in that comment should take care of everything.  The only thing that it doesn't do is show the banner on goal deletion.  In these cases patrons would see the banner the next time that they visit My Books.

Let me know if you have any questions, or if there's any way that I can make this code easier to maintain.

File: openlibrary/plugins/openlibrary/js/check-ins/index.js [hunk 3]
```diff
@@ -520,19 +522,23 @@ function addGoalSubmissionListener(submitButton) {
...
                         goalContainer.remove()
                         chipGroup.classList.remove('hidden')
+                        location.reload()
                     } else {
-                        const progressComponent = modal.closest('.reading-goal-progress')
-                        updateProgressComponent(progressComponent, Number(formData.get('goal')))
+                        const progressComponent = document.querySelector('.reading-goal-progress')
+                        if (progressComponent) {
+                            updateProgressComponent(progressComponent, Number(formData.get('goal')))
+                        }
```
Comment (mekarpeles) [inline openlibrary/plugins/openlibrary/js/check-ins/index.js:541]:
```suggestion
                        const progressComponent = modal.closest('.reading-goal-progress')
                        updateProgressComponent(progressComponent, Number(formData.get('goal')))
```

File: openlibrary/plugins/openlibrary/js/check-ins/index.js [hunk 4]
```diff
@@ -520,19 +522,23 @@ function addGoalSubmissionListener(submitButton) {
                     modal.close()
                 }
 
-                const yearlyGoalSection = modal.closest('.yearly-goal-section')
-                if (formData.get('is_update')) {  // Progress component exists on page
-                    const goalInput = form.querySelector('input[name=goal]')
+                const progressContainerExists = formData.get('is_update')
+                const yearlyGoalSection = document.querySelector('.yearly-goal-section')
+                const goalInput = progressContainerExists ? document.querySelector('input[name=goal]') : false
+                if (goalInput) {  // Progress component exists on page
```
Comment (mekarpeles) [inline openlibrary/plugins/openlibrary/js/check-ins/index.js:528]:
```suggestion
const yearlyGoalSection = modal.closest('.yearly-goal-section')
                if (formData.get('is_update')) {  // Progress component exists on page
                    const goalInput = form.querySelector('input[name=goal]')
```

File: openlibrary/plugins/openlibrary/js/check-ins/index.js [hunk 5]
```diff
@@ -520,7 +522,7 @@ function addGoalSubmissionListener(submitButton) {
                     modal.close()
                 }
 
-                const yearlyGoalSection = modal.closest('.yearly-goal-section')
+const yearlyGoalSection = modal.closest('.yearly-goal-section')
                 if (formData.get('is_update')) {  // Progress component exists on page
```
Comment (mekarpeles) [inline openlibrary/plugins/openlibrary/js/check-ins/index.js:526]:
```suggestion
                const yearlyGoalSection = modal.closest('.yearly-goal-section')
                if (formData.get('is_update')) {  // Progress component exists on page
```

**Review by jimchamp — APPROVED** [review]
Lgtm!  Thanks for your patience.

**Pull request merged**